samples/modelbuilder/ObjectDetection_StopSigns/StopSignDetectionML.Model/ConsumeModel.cs

```diff
@@ -0,0 +1,63 @@
+// This file was auto-generated by ML.NET Model Builder. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using Microsoft.ML;
+using StopSignDetectionML.Model;
+using System.Net.Http;
+using Microsoft.ML.Trainers;
+using System.Web;
+using System.Net;
+
+namespace StopSignDetectionML.Model
+{
+    public class ConsumeModel
+    {
+        private static Lazy<PredictionEngine<ModelInput, ModelOutput>> PredictionEngine = new Lazy<PredictionEngine<ModelInput, ModelOutput>>(CreatePredictionEngine);
+
+        private static WebClient client = new WebClient();
+
+        public static string MLNetModelPath = Path.GetFullPath(@"../../../../StopSignDetectionML.Model/MLModel.zip");
+
+        // For more info on consuming ML.NET models, visit https://aka.ms/mlnet-consume
+        // Method for consuming model in your app
+        public static ModelOutput Predict(ModelInput input)
+        {
+            ModelOutput result = PredictionEngine.Value.Predict(input);
+            return result;
+        }
+
+        public static PredictionEngine<ModelInput, ModelOutput> CreatePredictionEngine()
+        {
+            // Create new MLContext
+            MLContext mlContext = new MLContext();
+
+            // Download the model
+            if (!File.Exists(MLNetModelPath))
+            {
+                DownloadModel();
+            }
+
+            // Load model & create prediction engine
+            ITransformer mlModel = mlContext.Model.Load(MLNetModelPath, out var modelInputSchema);
+
+            var predEngine = mlContext.Model.CreatePredictionEngine<ModelInput, ModelOutput>(mlModel);
+
+             return predEngine;
+        }
+
+        private static void DownloadModel()
```

Comment:
Maybe we can move this into the project file instead using https://docs.microsoft.com/en-us/visualstudio/msbuild/downloadfile-task?view=vs-2019.

```
  <ItemGroup>
    <ModelFileUrl Include="https://mlpublicassets.blob.core.windows.net/samples/object-detection/MLModel.zip" />
  </ItemGroup>

  <Target Name="DownloadModelFiles" BeforeTargets="Build">
    <DownloadFile SourceUrl="@(ModelFileUrl)" SkipUnchangedFiles="true" DestinationFolder="$(OutDir)"/>
  </Target>
```

That way you don't need to do this at runtime and it's closer to what we would do if we could check in the model.

Reply:
Also, we might want to use an aka.ms link in case we need to change the storage location in the future.

Reply:
Fixed with https://github.com/dotnet/machinelearning-samples/pull/882/commits/d559ca315873ee6fad927f449f0a690e3d562f8b